### ISSUE TITLE
Supported critical extensions is hard-coded and cannot be changed by user code

### ISSUE THREAD

**Issue by johnwason** [digitalbazaar/forge#358]
I am using forge with a custom certificate application that has custom critical extensions used to identity the type of certificate.  The extensions are checked in the tls verify callback. Currently the extension checking has the supported critical extensions hard coded in x509.js:

    var se = {
        keyUsage: true,
        basicConstraints: true
      };
      for(var i = 0; error === null && i < cert.extensions.length; ++i) {
        var ext = cert.extensions[i];
        if(ext.critical && !(ext.name in se)) {
          error = {
            message:
              'Certificate has an unsupported critical extension.',
            error: pki.certificateError.unsupported_certificate
          };
        }
      }

Can an option be added to ignore critical extension errors, or can the "se" variable be made a global module variable like "oids" that can be modified?

**Comment by dlongley** [comment]
Yes, we should make this extensible, just haven't had time to design something that will work well here. PRs are welcome.

**Comment by johnwason** [comment]
Would it make sense to have an option `ignoreCriticalExtensions` for `createConnection`?  For instance,

    var client = forge.tls.createConnection({
         ignoreCriticalExtensions: true,
         /* other options */
    }

I can probably create a PR to implement this.

**Comment by thirtified** [comment]
👍 to make it possible to work with additional or custom critical extensions. E.g. Open Banking UK certificates have the Extended Key Usage extension marked as critical and therefore currently fail verification with node-forge.

**Comment by thirtified** [comment]
See also #500.
